### DISCUSSION TITLE
Inconsistent UTC date handling in SfDate

### DISCUSSION THREAD

**Discussion by RAWeber** [jsforce/jsforce#1494] [Q&A]
When using SfDate, if a time value is given with the date it is treated as UTC, however, if no time value is given it is treated as local time.

This can be seen in the `SfDate.parseDate` function:
https://github.com/jsforce/jsforce/blob/b6bebda5a5b9d963413649f4553a6cd382bed4a9/lib/date.js#L84-L104

As such, when using `SfDate. toDateTimeLiteral` dates are returned with a local time offset if no time is given, but in utc if a time is specified.

**Comment by chexxor** [comment]
If you want to change existing functionality, you'll need to give a compelling reason. If this is changed, applications could break for a number of people on the order of the number of current users of JSforce, and the only benefit is one person. It seems better to have the one person work around it by adjusting the date values they pass into JSforce.

On the other hand, are you requesting documentation around date and datetime handling be improved? If so, you'll need to point to places in the current docs which you'd like to be changed. Or you'll need to submit a PR to improve docs yourself, because you're the most qualified person to do that.

**Comment by cristiand391** [comment]
moved this to GH discussions...

tbh, I'm not verify familiar with this so please let me know if I'm wrong:

defaulting to local time is technically correct but discouraged:
https://en.wikipedia.org/wiki/ISO_8601#Local_time_(unqualified)

maybe documenting this so users are aware of how dates without time info are handled is enough? as mentioned above, making the change now would break users.
